src/lib.rs

```diff
@@ -1,12 +1,1184 @@
+    pub const DeleteObject: Value = 0x100B;
+    pub const SendObjectInfo: Value = 0x100C;
+    pub const SendObject: Value = 0x100D;
+    pub const InitiateCapture: Value = 0x100E;
+    pub const FormatStore: Value = 0x100F;
+    pub const ResetDevice: Value = 0x1010;
+    pub const SelfTest: Value = 0x1011;
+    pub const SetObjectProtection: Value = 0x1012;
+    pub const PowerDown: Value = 0x1013;
+    pub const GetDevicePropDesc: Value = 0x1014;
+    pub const GetDevicePropValue: Value = 0x1015;
+    pub const SetDevicePropValue: Value = 0x1016;
+    pub const ResetDevicePropValue: Value = 0x1017;
+    pub const TerminateOpenCapture: Value = 0x1018;
+    pub const MoveObject: Value = 0x1019;
+    pub const CopyObject: Value = 0x101A;
+    pub const GetPartialObject: Value = 0x101B;
+    pub const InitiateOpenCapture: Value = 0x101C;
+    
+    pub fn name(v: Value) -> Option<&'static str> {
+        match v {
+            Undefined => Some("Undefined"),
+            GetDeviceInfo => Some("GetDeviceInfo"),
+            OpenSession => Some("OpenSession"),
+            CloseSession => Some("CloseSession"),
+            GetStorageIDs => Some("GetStorageIDs"),
+            GetStorageInfo => Some("GetStorageInfo"),
+            GetNumObjects => Some("GetNumObjects"),
+            GetObjectHandles => Some("GetObjectHandles"),
+            GetObjectInfo => Some("GetObjectInfo"),
+            GetObject => Some("GetObject"),
+            GetThumb => Some("GetThumb"),
+            DeleteObject => Some("DeleteObject"),
+            SendObjectInfo => Some("SendObjectInfo"),
+            SendObject => Some("SendObject"),
+            InitiateCapture => Some("InitiateCapture"),
+            FormatStore => Some("FormatStore"),
+            ResetDevice => Some("ResetDevice"),
+            SelfTest => Some("SelfTest"),
+            SetObjectProtection => Some("SetObjectProtection"),
+            PowerDown => Some("PowerDown"),
+            GetDevicePropDesc => Some("GetDevicePropDesc"),
+            GetDevicePropValue => Some("GetDevicePropValue"),
+            SetDevicePropValue => Some("SetDevicePropValue"),
+            ResetDevicePropValue => Some("ResetDevicePropValue"),
+            TerminateOpenCapture => Some("TerminateOpenCapture"),
+            MoveObject => Some("MoveObject"),
+            CopyObject => Some("CopyObject"),
+            GetPartialObject => Some("GetPartialObject"),
+            InitiateOpenCapture => Some("InitiateOpenCapture"),
+            _ => None,
+        }
+    }
+}
+
+/// An error in a PTP command
+#[derive(Debug)]
+pub enum Error {
+    /// PTP Responder returned a status code other than Ok, either a constant in StandardResponseCode or a vendor-defined code
+    Response(u16),
+    
+    /// Device did not respond within the timeout interval
+    Timeout,
+    
+    /// Device has been disconnected
+    NoDevice,
+    
+    /// Data received was malformed
+    Malformed(String),
+    
+    /// Another libusb error
```

Comment:
Should NoDevice still be part of the Usb enum? The R10C code treats it specially but it feels like a leaky abstraction to make the ptp crate do so too.